### PR TITLE
Fix/gpg agent forwarding arg list

### PR DESCRIPTION
**Pull request by pascalbreuninger**
In cases where the gpg public key exceeds ARG_MAX of the workspace
shelll gpg forwarding fails because the shell rejects the command.
We now fetch the private key through the workspace credentials server
before decoding and passing to `gpg`

### CODE REVIEW

File: cmd/agent/workspace/setup_gpg.go [hunk 1]
```diff
@@ -53,8 +53,14 @@ func NewSetupGPGCmd(flags *flags.GlobalFlags) *cobra.Command {
 func (cmd *SetupGPGCmd) Run(ctx context.Context, log log.Logger) error {
 	log.Debugf("Initializing gpg-agent forwarding")
 
-	log.Debugf("Decoding input public key")
-	publicKey, err := base64.StdEncoding.DecodeString(cmd.PublicKey)
+	log.Debugf("Fetching public key")
+	rawPublicKeys, err := getPublicKeys(log)
+	if err != nil {
+		return err
```
Comment (bkneis) [inline cmd/agent/workspace/setup_gpg.go:59]: should we be logging this error? also on line 65

File: pkg/agent/tunnelserver/tunnelserver.go [hunk 2]
```diff
@@ -264,6 +266,17 @@ func (t *tunnelServer) LoftConfig(ctx context.Context, message *tunnel.Message)
 	return &tunnel.Message{Message: string(out)}, nil
 }
 
+func (t *tunnelServer) GPGPublicKeys(ctx context.Context, message *tunnel.Message) (*tunnel.Message, error) {
```
Comment (bkneis) [inline pkg/agent/tunnelserver/tunnelserver.go:269]: docblock explaining what this function is doing would be nice

File: pkg/credentials/request.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+		raw, err := io.ReadAll(response.Body)
+		if err != nil {
+			return err
+		}
+
+		// has the request succeeded?
+		if response.StatusCode != http.StatusOK {
+			log.Errorf("Error calling %s (%d): %w", endpoint, response.StatusCode, string(raw))
+			return err
```
Comment (bkneis) [inline pkg/credentials/request.go:47]: err might be nil here, maybe assing the error above in the log then log it and return

File: pkg/credentials/request.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+		}
+
+		out = raw
```
Comment (bkneis) [inline pkg/credentials/request.go:50]: you could simply reuse out instead of temp var raw?

**Review by bkneis — APPROVED** [review]
LGTM, just a couple small comments

File: pkg/agent/tunnelserver/tunnelserver.go (hunk 2 above)
Comment (pascalbreuninger) [reply pkg/agent/tunnelserver/tunnelserver.go:269]: Let's do that another time when we document all of the GRPC endpoints :)

File: pkg/credentials/request.go (hunk 3 above)
Comment (pascalbreuninger) [reply pkg/credentials/request.go:47]: good catch, thanks. Err will always be nil here so we now just return a new one

**Comment by pascalbreuninger** [comment]
Abandonding in favor of #1336

**Pull request closed without merging**